selendroid-standalone/src/main/java/io/selendroid/standalone/android/AndroidSdk.java

```diff
@@ -124,25 +108,91 @@ public static String androidJar() {
   }
 
   public static File androidSdkFolder() {
-    String platformsRootFolder = androidHome() + File.separator + "platforms";
-    File platformsFolder = new File(platformsRootFolder);
-
     if (sAndroidSdkVersion != null) {
-      return new File(platformsFolder, sAndroidSdkVersion);
-    } else {
-      return findLatestAndroidPlatformFolder(
-        platformsFolder,
-        "No installed Android APIs have been found.");
+      return new File(platformsFolder(), sAndroidSdkVersion);
+    }
+    return findLatestAndroidPlatformFolder();
+  }
+
+  private static File platformsFolder() {
+    return new File(androidHome(), "platforms");
+  }
+
+  public static File findLatestAndroidPlatformFolder() {
+    return findLatestAndroidPlatformFolder(platformsFolder());
+  }
+
+  public static File findLatestAndroidPlatformFolder(File platformsFolder) {
+    File[] platformVersions = platformsFolder.listFiles(new FileFilter() {
+      @Override
+      public boolean accept(File file) {
+        return file.getName().matches("android-\\d+$");
```

Comment:
My inclination would be to make this regular expression a local variable, add a capture group for the integer, and then reuse that on lines 142 to get the version. Relying on a regular expression here and split below seems like it could create problems in the future if someone only updates this in one place.